docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+readable save files. The format in which the `LocalDateTime` objects are converted to can be found in the `DateTimeFormatter` object in the 
+`FileCommand` class.
+
+![Save sequence diagram](images/DeveloperGuide/StorageSaveSequenceDiagram.png)
+_Figure 4.7-3: Sequence diagram for saving of user note data_
+
+##### Implementation - Note Archiving
+
+The `archive` command immediately moves all currently loaded notes into a newly generated text file in the `Archives` directory found within the 
+_LifEasierSaves_ directory. If no `Archives` directory is found, it is automatically created. Archive save files are automatically named as the 
+current date in the **DD-MM-YY** format, and the time the archive command was run in the **HH:MM** format, separated by a **T**. The current save 
+file for notes will be automatically cleared with the `clearSaveFile()` command found in the `FileCommand` class, and the current `noteList` is 
+cleared. Archived notes will not be read by the program anymore and any changes can be made to the created archive save file.
+
+The `archive` command checks for the size of the current `noteList` before execution, and as such, when an empty `noteList` is detected, 
+the archiving process will not be started.
+
+##### Design Considerations
+
+In order to ensure users get the best hassle free and user-friendly experience while using **LifEasier**, saves are automatically done after any change 
+that affects any user added tasks and notes. While the constant clearing and rewriting of the save data whenever a change occurs may affect performance 
+when the save files get larger, it was decided that the convenience of an automatic saving system outweighs the performance costs, and the assurance 
+granted to users that their data is constantly saved without needing their manual intervention.
+
+Saves were also designed to be stored in simple plain text and easily accessible to users to allow experienced users to modify 
+the save files directly and easily, if required. 
+
+In the event of corrupted or missing data, the `storage` component defends and protects the app from potential issues that might arise from 
+reading in this data by throwing exceptions to stop any further data reading. Any data read up to that point is untouched, and the app will 
+continue to run as per normal. **Manual intervention from the user** is required to remove improperly formatted and/or missing data.  
+
+### 4.8 Displaying Schedule (Johannine)
+
+The `displaySchedule` command presents the `TaskList` contents in a timetable format, given that it is specified to 
+display the full week. Otherwise, it displays the current day’s schedule in a list form, with the `Task` items sorted 
+by date.
+
+![Figure 4.8-1](images/DeveloperGuide/Figure 4.8-1.png)
+_Figure 4.8-1: Sequence diagram for displaying week or day schedule_
```

Comment:
Should the activation bar end before the self-activation bar returns to itself? 

<img width="855" alt="Screenshot 2020-10-29 at 12 16 53 PM" src="https://user-images.githubusercontent.com/57402349/97525025-e3ca8080-19e0-11eb-905b-8a8afdb0a0b3.png">